src/web/app/components/session-edit-form/session-edit-form.component.spec.ts

```diff
@@ -53,4 +67,312 @@ describe('SessionEditFormComponent', () => {
+    component.model.customResponseVisibleDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
+    const maxTimeForSessionVisible = component.maxTimeForSessionVisible;
+    expect(maxTimeForSessionVisible).toEqual(component.model.customResponseVisibleTime);
+  });
+
+  it('should return the default time format if response visible setting is not recognized', () => {
+    const maxTimeForSessionVisible = component.maxTimeForSessionVisible;
+    expect(maxTimeForSessionVisible).toEqual(getDefaultTimeFormat());
+  });
+
+  it('should return submissionStartDate as the minimum date for response visibility'
+    + ' when session visible setting is AT_OPEN', () => {
+    component.model.sessionVisibleSetting = SessionVisibleSetting.AT_OPEN;
+    component.model.submissionStartDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
+    const minDateForResponseVisible = component.minDateForResponseVisible;
+    expect(minDateForResponseVisible).toEqual(component.model.submissionStartDate);
+  });
+
+  it('should return customSessionVisibleDate as the minimum date for response visibility '
+      + 'when session visible setting is CUSTOM', () => {
+    component.model.sessionVisibleSetting = SessionVisibleSetting.CUSTOM;
+    component.model.submissionStartDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
+    component.model.customSessionVisibleDate =
+        service.getDateInstance(moment().tz(component.model.timeZone));
+    const minDateForResponseVisible = component.minDateForResponseVisible;
+    expect(minDateForResponseVisible).toEqual(component.model.customSessionVisibleDate);
+  });
+
+  it('should return the default date format if session visible setting is not recognized', () => {
+    const minDateForResponseVisible = component.minDateForResponseVisible;
+    expect(minDateForResponseVisible).toEqual(getDefaultDateFormat());
+  });
+
+  it('should return submissionStartTime as the minimum time for response visibility '
+      + 'when session visible setting is AT_OPEN', () => {
+    component.model.sessionVisibleSetting = SessionVisibleSetting.AT_OPEN;
+    component.model.submissionStartTime =
+        service.getTimeInstance(moment().tz(component.model.timeZone));
+    const minTimeForResponseVisible = component.minTimeForResponseVisible;
+    expect(minTimeForResponseVisible).toEqual(component.model.submissionStartTime);
+  });
+
+  it('should return customSessionVisibleTime as the minimum time for response visibility '
+      + 'when session visible setting is CUSTOM', () => {
+    component.model.sessionVisibleSetting = SessionVisibleSetting.CUSTOM;
+    component.model.submissionStartTime =
+        service.getTimeInstance(moment().tz(component.model.timeZone));
+    component.model.customSessionVisibleTime =
+        service.getTimeInstance(moment().tz(component.model.timeZone));
```

Comment:
same here, let's make the customSessionVisibleTime different from submissionStartTime